config/mclag.py

```diff
@@ -121,34 +124,42 @@ def mclag(ctx):
-
-    db = ctx.obj['db']
+    if ADHOC_VALIDATION:
+        if not mclag_domain_id_valid(domain_id):
+            ctx.fail("{} invalid domain ID, valid range is 1 to 4095".format(domain_id))  
+        if not is_ipv4_addr_valid(source_ip_addr):
+            ctx.fail("{} invalid local ip address".format(source_ip_addr))
+        if not is_ipv4_addr_valid(peer_ip_addr):
+            ctx.fail("{} invalid peer ip address".format(peer_ip_addr))
+
+    db = ValidatedConfigDBConnector(ctx.obj['db'])
     fvs = {}
     fvs['source_ip'] = str(source_ip_addr)
     fvs['peer_ip'] = str(peer_ip_addr)
-    if peer_ifname is not None:
-        if (peer_ifname.startswith("Ethernet") is False) and (peer_ifname.startswith("PortChannel") is False):
-            ctx.fail("peer interface is invalid, should be Ethernet interface or portChannel !!")
-        if (peer_ifname.startswith("Ethernet") is True) and (check_if_interface_is_valid(db, peer_ifname) is False):
-            ctx.fail("peer Ethernet interface name is invalid. it is not present in port table of configDb!!")
-        if (peer_ifname.startswith("PortChannel")) and (is_portchannel_name_valid(peer_ifname) is False):
-            ctx.fail("peer PortChannel interface name is invalid !!")
-        fvs['peer_link'] = str(peer_ifname)
+    if ADHOC_VALIDATION:
+        if peer_ifname is not None:
+            if (peer_ifname.startswith("Ethernet") is False) and (peer_ifname.startswith("PortChannel") is False):
+                ctx.fail("peer interface is invalid, should be Ethernet interface or portChannel !!")
+            if (peer_ifname.startswith("Ethernet") is True) and (check_if_interface_is_valid(db, peer_ifname) is False):
+                ctx.fail("peer Ethernet interface name is invalid. it is not present in port table of configDb!!")
+            if (peer_ifname.startswith("PortChannel")) and (is_portchannel_name_valid(peer_ifname) is False):
+                ctx.fail("peer PortChannel interface name is invalid !!")
+    fvs['peer_link'] = str(peer_ifname)
     mclag_domain_keys = db.get_table('MCLAG_DOMAIN').keys()
     if len(mclag_domain_keys) == 0:
-        db.set_entry('MCLAG_DOMAIN', domain_id, fvs)
+        try:
+            db.set_entry('MCLAG_DOMAIN', domain_id, fvs)
+        except ValueError as e:
+            ctx.fail("Invalid ConfigDB. Error: {}".format(e))
     else:
+        domain_id = str(domain_id)
         if domain_id in mclag_domain_keys:
-            db.mod_entry('MCLAG_DOMAIN', domain_id, fvs)
-        else: 
-            ctx.fail("only one mclag Domain can be configured. Already one domain {} configured ".format(mclag_domain_keys[0]))  
+            try:
+                db.mod_entry('MCLAG_DOMAIN', domain_id, fvs)
+            except ValueError as e:
+                ctx.fail("Invalid ConfigDB. Error: {}".format(e))
+        else:
+            ctx.fail("only one mclag Domain can be configured. Already one domain {} configured ".format(list(mclag_domain_keys)[0]))
```

Comment:
>list [](http://example.com/codeflow?start=105&length=4)

Just wondering whether this is a bug fix?

Reply:
Yes. Without the change, we get this error
```
  File "/usr/local/lib/python3.9/dist-packages/config/mclag.py", line 161, in add_mclag_domain
    print(mclag_domain_keys[0])
TypeError: 'dict_keys' object is not subscriptable 
```

Reply:
Thanks for your contribution to fix a bug in codebase!